source/common/stats/allocator_impl.h

```diff
@@ -93,6 +103,15 @@ class AllocatorImpl : public Allocator {
   std::vector<GaugeSharedPtr> deleted_gauges_ ABSL_GUARDED_BY(mutex_);
   std::vector<TextReadoutSharedPtr> deleted_text_readouts_ ABSL_GUARDED_BY(mutex_);
 
+  template <typename StatType>
+  using StatPointerSet = absl::flat_hash_set<StatType*, absl::Hash<StatType*>>;
+  // Stat pointers that participate in the flush to sink process.
+  StatPointerSet<Counter> sinked_counters_ ABSL_GUARDED_BY(mutex_);
+  StatPointerSet<Gauge> sinked_gauges_ ABSL_GUARDED_BY(mutex_);
+  StatPointerSet<TextReadout> sinked_text_readouts_ ABSL_GUARDED_BY(mutex_);
+
+  // Predicates used to filter stats to be flushed.
+  const SinkPredicates* sink_predicates_ = nullptr;
```

Comment:
suggest using `envoy/common/optref.h` per envoy style. It's basically line-for-line the same but is slightly more obvious in intent.

However, I think this suggest is overridden by a different idea: can we put the ownership of the predicate in here and remove it from server.h? If we need to access it from the server we could provide an accessor in the Allocator, which would return an OptRef. Scanning the PR I could not find where you'd need to use this from the server object.